janusgraph-server/src/main/java/org/janusgraph/graphdb/tinkerpop/gremlin/server/auth/JanusGraphAbstractAuthenticator.java

```diff
@@ -114,8 +115,33 @@ public void setup(final Map<String,Object> config) {
         }
 
         final String defaultUser = config.get(CONFIG_DEFAULT_USER).toString();
-        if (credentialStore.findUser(defaultUser) == null) {
-            credentialStore.createUser(defaultUser, config.get(CONFIG_DEFAULT_PASSWORD).toString());
+        if (!userExists(defaultUser)) {
+            createUser(defaultUser, config.get(CONFIG_DEFAULT_PASSWORD).toString());
+        }
+    }
+
+    protected Vertex findUser(final String username) {
+        credentials.tx().rollback();
+        final GraphTraversal<Vertex,Vertex> t = credentials.users(username);
+        final Vertex v = t.hasNext() ? t.next() : null;
+        if (t.hasNext()) throw new IllegalStateException(String.format("Multiple users with username %s", username));
+        return v;
+    }
+
+    private boolean userExists(final String username) {
+        credentials.tx().rollback();
+        return credentials.users(username).hasNext();
+    }
+
+    private void createUser(final String username, final String password) {
+        credentials.tx().rollback();
```

Comment:
Same here

Reply:
Again, same answer. [Here](https://github.com/apache/tinkerpop/blob/ea1e5a323a57d265bc3e588c6e40a30dfc2e8c42/gremlin-groovy/src/main/java/org/apache/tinkerpop/gremlin/groovy/jsr223/dsl/credential/CredentialGraph.java#L69) you can find `CredentialGraph.createUser()` from TinkerPop 3.3.z for reference.